### DISCUSSION TITLE
micropython-lib already built in

### DISCUSSION THREAD

**Discussion by osto123** [micropython/micropython#16467] [Libraries & Drivers]
Hello, I have just started with micropython with ESP32 and THonny IDE and so far no problems. I still have a question regarding micropython-lib that seems to be built in by default. So far I have used libraries like "mqtt.simple" and "requests" However the "docs.micropython.org" libraries listing does not mention anything about those libraries. Is it that the documentation is out of date or what am I not understanding because reading "docs.micropython.org" or "micropython-lib" github site I get assumption that those libraries should be installed using mip, but obviously they are already built in. Is there something foundational that I dont undestand on this... Thank you.

**Comment by davefes** [comment]
Maybe:
https://github.com/micropython/micropython-lib/tree/master/micropython/umqtt.simple

**Reply by jonnor** [reply]
Only a few packages are included by default. The rest you install via "mip". See https://docs.micropython.org/en/latest/reference/packages.html

**Reply by scruss** [reply]
And for finding the short names of micropython-lib packages, I always reach for [mim](https://mim.oliverrobson.tech/)

**Comment by Josverl** [comment]
Also, to check which modules are built-in (frozen) you can run the following in the REPL
```
>>> help('modules')
```
